bin/node/src/commands/store.rs

```diff
@@ -0,0 +1,273 @@
+use std::{
+    fs::File,
+    io::Write,
+    path::{Path, PathBuf},
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+use anyhow::Context;
+use miden_lib::{AuthScheme, account::faucets::create_basic_fungible_faucet, utils::Serializable};
+use miden_node_store::{genesis::GenesisState, server::Store};
+use miden_node_utils::{crypto::get_rpo_random_coin, grpc::UrlExt};
+use miden_objects::{
+    Felt, ONE,
+    account::{AccountFile, AccountIdAnchor, AuthSecretKey},
+    asset::TokenSymbol,
+    crypto::dsa::rpo_falcon512::SecretKey,
+};
+use rand::{Rng, SeedableRng};
+use rand_chacha::{ChaCha20Rng, ChaChaRng};
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+use super::{ENV_ENABLE_OTEL, ENV_STORE_DIRECTORY, ENV_STORE_URL};
+
+#[derive(clap::Subcommand)]
+pub enum StoreCommand {
+    /// Dumps the default genesis configuration to stdout.
+    ///
+    /// Use this as a starting point to modify the genesis data for `bootstrap`.
+    DumpGenesis,
+
+    /// Bootstraps the blockchain database with the genesis block.
+    ///
+    /// This populates the genesis block's data with the accounts and data listed in the
+    /// configuration file.
+    ///
+    /// Each generated genesis account's data is also written to disk. This includes the private
+    /// key which can be used to create transactions for these accounts.
+    ///
+    /// See also: `dump-genesis`
+    Bootstrap {
+        /// Genesis configuration file.
+        ///
+        /// If not provided the default configuration is used.
+        #[arg(long, value_name = "FILE")]
+        config: Option<PathBuf>,
+        /// Directory in which to store the database and raw block data.
+        #[arg(long, env = ENV_STORE_DIRECTORY, value_name = "DIR")]
+        data_directory: PathBuf,
+        // Directory to write the account data to.
+        #[arg(long, value_name = "DIR")]
+        accounts_directory: PathBuf,
+    },
```

Comment:
It is not clear to me that these commands should be store-specific. Currently the do affect only the store component, but in the future, bootstrapping process may affect multiple components. Maybe we could move it to the root so that the users could run something like:
```
miden-node bootstrap [OPTIONS] --data-directory <DIR> --accounts-directory <DIR>
```

Reply:
hmmm - not quite sure what an example could be, and actually, thinking about this more, it does feel like the store component should have its own `bootstrap` command. But I do think it may be good to have one "global" `bootstrap` command which would currently just execute store's initialization process. So, both of these will currently do the same:
```
miden-node bootstrap [OPTIONS] --data-directory <DIR> --accounts-directory <DIR>
miden-node store bootstrap [OPTIONS] --data-directory <DIR> --accounts-directory <DIR>
```
But in the future the second one may be a subset of the first one.

Reply:
I would prefer if we have a single canonical way of doing one thing, at least until there is a stronger indication that we need a combination of things.

I _could_ see that we may want a quick-start option to spin up a node but its possible we would want that to form part of the "bundled" options then?

Reply:
Would you be happy with

```
miden-node store boostrap
           store start
           
miden-node bundled bootstrap
           bundled start
```

where `bundled` is the current `node` -- I don't think this is a great name, but `node` is also.. strange. I'm wary of dumping the full node commands in the root "folder" so to speak.

Reply:
> miden-node store boostrap
>            store start
>            
> miden-node bundled bootstrap
>            bundled start

Yep, let's go with this. I do think not having the "bundled" word would make it a bit easier for the user - but the difference is probably marginal.

Reply:
Node alternate word dump

```
composite
combined
bundle
pack
in-process
full-node
full
all
```